contracts/staking/StakingPoolLib.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+library StakingPoolLib {
+    enum RewardType {
+        NONE,
+        FIXED,
+        FLOATING
+    }
+
+    struct Reward {
+        address tokens;
```

Comment:
We should enforce that this is an ERC20, or a sub-interface that want to require of our reward tokens.

Reply:
> tokens should be just token.

The Bonds contracts call this `tokens` for singular ERC-20, since there are plural tokens within a users balance, so kept with this naming convention

cc @CjHare Can probably explain it better

Reply:
> We should enforce that this is an ERC20, or a sub-interface that want to require of our reward tokens.

do you mean with ERC-165 or similar?